src/cljs/rems/application.cljs

```diff
@@ -224,11 +225,38 @@
                   edit-application
                   (fn [response]
                     (rf/dispatch [::fetch-application (:application/id application)])
-                    (apply (handle-validations! description application on-success) [response]))))
+                    (apply (handle-validations! description application {:on-success on-success}) [response]))))
    {:db (-> db
             (assoc-in [::edit-application :validation :errors] nil)
             (assoc-in [::edit-application :validation :warnings] nil))}))
 
+
+(rf/reg-event-db ::set-autosaving (fn [db [_ value]] (assoc db ::autosaving value)))
+(rf/reg-sub ::autosaving (fn [db _] (::autosaving db)))
+
+(rf/reg-event-fx
+ ::autosave-application
+ (fn [{:keys [db]} [_]]
+   (if (-> db :config :enable-autosave)
+     (let [application (-> db ::application :data)
+           edit-application (::edit-application db)
+           description [text :t.form/autosave]]
+       (save-draft! description
+                    application
+                    edit-application
+                    (fn [response]
+                      (rf/dispatch [::fetch-application (:application/id application) false])
+                      (apply (handle-validations! description application {:on-success #(do (rf/dispatch [::set-autosaving false])
+                                                                                            (flash-message/show-quiet-success! :actions-form-flash [text :t.form/autosave-confirmed]))
+                                                                           :default-success? false
+                                                                           :focus? false})
+                             [response])))
+       {:db (-> db
+                (assoc ::autosaving true)
+                (assoc-in [::edit-application :validation :errors] nil)
+                (assoc-in [::edit-application :validation :warnings] nil))})
+     {})))
```

Comment:
one idea for shortening the function, if fetching application is not critical to perform prior to validations

```clojure
(->> (handle-validations! description
                          application
                          {:on-success (fn []
                                         (rf/dispatch [::set-autosaving false])
                                         (flash-message/show-quiet-success! :actions-form-flash
                                                                            [text :t.form/autosave-confirmed])
                                         ;; equal to (rf/dispatch [::fetch-application (:application/id application) false])
                                         (reload! (:application/id application) false))
                           :default-success? false
                           :focus? false})
     (save-draft! description application edit-application))
```

Reply:
This is now included in #3037